src/hooks/useBlockActivityData.tsx

```diff
@@ -0,0 +1,17 @@
+import { useQuery } from "react-query";
+import instance from "../axios/instance";
+
+import { useChannelHash } from "./useChannelHash";
+
+const useBlockActivityData = (blockProp: string) => {
+  const { channelHash } = useChannelHash();
+  const { data, isLoading } = useQuery("blocks-activity", async () => {
+    const response = await instance.get(`/${blockProp}/${channelHash}`);
```

Comment:
I think it'd be better to just put blockActivity there as a string instead of as a prop.